slicedimage/io.py

```diff
@@ -25,7 +25,9 @@ def infer_backend(baseurl, allow_caching=True):
     elif parsed.scheme == "file":
         backend = DiskBackend(parsed.path)
     else:
-        raise ValueError("Unable to infer backend for url {}".format(baseurl))
+        raise ValueError(
+                "Unable to infer backend for url {}, please verify that the file or URL was correctly "
+                "specified.".format(baseurl))
```

Comment:
That's good, but I'd like one of two things in addition so that  a user can fix the problem, based on the error message. 

In this case, I've passed a path that doesn't exist. 

So, either the error message should define what a valid URL scheme is, or the docs should. Without that I would have trouble fixing the error without knowledge of the package.